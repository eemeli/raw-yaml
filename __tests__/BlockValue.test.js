import { commonTests, testParse } from './common'

describe("parse >block", () => {
  const block = '      #multiline\n  \n      \tblock'
  for (const name in commonTests) {
    const props = Object.assign({ str: `>\n${block}`, expected: block }, commonTests[name])
    if (props.post && props.post[0] !== '\n') {
      props.pre = `  ${props.pre}`
      props.post = props.post.replace(/^\s?/, '\n')
    }
    test(name, () => testParse(props))
  }
  test('literal with header', () => testParse({ pre: '\n- ', str: `|+2\n${block}`, expected: block, post: '\n' }))
})
