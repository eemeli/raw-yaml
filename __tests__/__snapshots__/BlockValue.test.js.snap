// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse >block bare 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 34,
    "start": 0,
  },
  "raw": "      #multiline
  
      	block",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 34,
    "start": 2,
  },
}
`;

exports[`parse >block complex mapping key 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 39,
    "start": 4,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 39,
    "start": 6,
  },
}
`;

exports[`parse >block flow seq value 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 39,
    "start": 4,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 39,
    "start": 6,
  },
}
`;

exports[`parse >block indented block 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 41,
    "start": 6,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 41,
    "start": 8,
  },
}
`;

exports[`parse >block literal with header 1`] = `
BlockValue {
  "blockStyle": "|+2",
  "range": Range {
    "end": 40,
    "start": 3,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_LITERAL",
  "valueRange": Range {
    "end": 40,
    "start": 7,
  },
}
`;

exports[`parse >block newline before & after 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 36,
    "start": 1,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 36,
    "start": 3,
  },
}
`;

exports[`parse >block seq value 1`] = `
BlockValue {
  "blockStyle": ">",
  "range": Range {
    "end": 37,
    "start": 2,
  },
  "raw": "      #multiline
  
      	block
",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 37,
    "start": 4,
  },
}
`;

exports[`parse >block with comment 1`] = `
BlockValue {
  "blockStyle": ">",
  "commentRange": Range {
    "end": 21,
    "start": 6,
  },
  "range": Range {
    "end": 55,
    "start": 3,
  },
  "raw": "      #multiline
  
      	block
",
  "rawComment": "comment # here!",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 55,
    "start": 22,
  },
}
`;

exports[`parse >block with props 1`] = `
BlockValue {
  "anchor": "anchor",
  "blockStyle": ">",
  "range": Range {
    "end": 51,
    "start": 2,
  },
  "raw": "      #multiline
  
      	block
",
  "tag": "tag!",
  "type": "BLOCK_FOLDED",
  "valueRange": Range {
    "end": 51,
    "start": 18,
  },
}
`;
